src/Orleans/Messaging/ProxiedMessageCenter.cs

```diff
@@ -287,6 +293,12 @@ public Message WaitMessage(Message.Categories type, CancellationToken ct)
         {
             try
```

Comment:
should add catch clauses for `ObjectDisposedException` and `InvalidOperationException` from the [MSDN docs]( https://msdn.microsoft.com/en-us/library/dd287085(v=vs.110).aspx).